implementations/rust/daemon/src/worker.rs

```diff
@@ -52,7 +57,24 @@ impl Worker {
         self.config.clone()
     }
 
-    pub fn poll(&self) -> bool {
+    fn receive_channel(&mut self, m: Message) -> Result<(), String> {
+        self.route = Some(m.return_route.clone());
+        Ok(())
+        // let resp_public_key = encode(&m.message_body);
+        // println!("Remote static public key: {}", resp_public_key);
+        // if let Some(rpk) = self.config.remote_public_key() {
+        //     if rpk == encode(&m.message_body) {
+        //         println!("keys agree");
+        //         return Ok(());
+        //     } else {
+        //         println!("keys conflict");
+        //         return Err("remote public key doesn't match expected, possible spoofing".into());
+        //     }
+        // }
+        // Ok(())
+    }
```

Comment:
does this introduce a regression? are we not validating the public key elsewhere if this is commented out?

Reply:
This is the sink, which publishes its public key. It's the source side that validates the public key.